packages/ses/src/module-load.js

```diff
@@ -507,18 +507,72 @@ const memoizedLoadWithErrorAnnotation = (
   return moduleLoading;
 };
 
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback SyncJob
+ * @param {...Args} args
+ * @returns {void}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback AsyncJob
+ * @param {...Args} args
+ * @returns {Promise<void>}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback EnqueueSyncJobFn
+ * @param {SyncJob<Args>} func
+ * @param {readonly [...Args]} args // the only way to make the types match
+ * @returns {void}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback EnqueueAsyncJobFn
+ * @param {AsyncJob<Args>} func
+ * @param {readonly [...Args]} args // the only way to make the types match
+ * @returns {void}
+ */
+
+/**
+ * @typedef SyncJobQueueResult
+ * @property {EnqueueSyncJobFn} enqueueJob
+ * @property {() => void} drainQueue
+ * @property {Error[]} errors
+ */
+
+/**
+ * @typedef AsyncJobQueueResult
+ * @property {EnqueueAsyncJobFn} enqueueJob
+ * @property {() => Promise<void>} drainQueue
+ * @property {Error[]} errors
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @typedef { Array<[SyncJob, readonly [...Args]]>} SyncJobs
+ */
+
+/**
+ * @param {Error[]} errors
+ * @returns {AsyncJobQueueResult}
+ */
 const asyncJobQueue = (errors = []) => {
   /** @type {Set<Promise<undefined>>} */
   const pendingJobs = new Set();
 
   /**
    * Enqueues a job that starts immediately but won't be awaited until drainQueue is called.
    *
-   * @template {any[]} T
-   * @param {(...args: T)=>Promise<*>} func
-   * @param {T} args
+   * @type {EnqueueAsyncJobFn}
    */
   const enqueueJob = (func, args) => {
+    if (!isArray(args)) {
```

Comment:
Why do I need to suffer runtime performance hits to convince TS an array is still an array?